### PR TITLE
Feature/131

### PR DESCRIPTION
**Pull request by LimNora**
Adding MLTrainer, MLVersionDeployer and MLOperationWaiter

### CODE REVIEW

File: backends/core/workers.py [hunk 1]
```diff
@@ -816,6 +828,109 @@ def _execute(self):
...
+class MLTrainer(MLWorker):
+  """Worker to train a ML model"""
+
+  PARAMS = [
+      ('project', 'string', True, '', 'ML project ID'),
+      ('jobDir', 'string', True, '',
+       'URI of folder to put output generated by AI platform (e.g. gs://bucket/folder)'),
+      ('packageUris', 'string', True, '', 
+        'URI of python package e.g. gs://bucket/folder/filename.tar.gz'),
+      ('trainFile', 'string', True, '',
```
Comment (Slony) [inline backends/core/workers.py]: Please, replace all the parameters that go into `agrs` in the API call with a single multi-line parameter.

File: backends/core/workers.py [hunk 2]
```diff
@@ -816,6 +828,109 @@ def _execute(self):
...
+            '--eval-steps',
+            str(self._params['evalSteps']),
+            '--verbosity', 
+            'DEBUG'
+          ],
+          'packageUris': self._params['packageUris'],
+          'region': 'europe-west1',
+          'jobDir': self._params['jobDir'], 
+          'runtimeVersion': self._params['runtimeVersion'],
+          'pythonModule': '%s.task' % (self._params['pythonModule'])
```
[the remark this answers is not in the document]
Comment (LimNora) [reply backends/core/workers.py]: Thanks! I addressed this.

File: backends/core/workers.py (hunk 1 above)
Comment (LimNora) [reply backends/core/workers.py]: I implemented this

**Comment by LimNora** [comment]
Thanks a lot for the comments. I used pylint and addressed the issues mentioned.

File: backends/core/workers.py [hunk 3]
```diff
@@ -822,6 +837,103 @@ def _execute(self):
...
+    job_name = '%s/jobs/%s' % (project_id, self._ml_job_id)
+    self._enqueue('MLWaiter', {'job_name': job_name}, 60)
+
+class MLVersionDeployer(MLWorker, StorageWorker):
+  """Worker to deploy ML Model Version"""
+
+  PARAMS = [
+      ('project', 'string', True, '', 'ML project ID'),
+      ('jobDir', 'string', True, '',
+       'URI of folder to put output generated by AI platform '
```
[the remark this answers is not in the document]
Comment (LimNora) [reply backends/core/workers.py]: Thanks! I implemented this.

File: backends/core/workers.py [hunk 4]
```diff
@@ -822,6 +837,103 @@ def _execute(self):
...
+       '(e.g. gs://bucket/folder)'),
+      ('modelName', 'string', True, '',
+       'Name of the ML model in Google Cloud AI Platform'),
+      ('versionName', 'string', True, '',
+       'Name of the version (Letters, numbers, underscores only. '
```
[the remark this answers is not in the document]
Comment (LimNora) [reply backends/core/workers.py]: Thanks! I implemented this.

File: backends/core/workers.py [hunk 5]
```diff
@@ -822,6 +837,103 @@ def _execute(self):
...
+  def _execute(self):
+    self._get_ml_job_id()
+
+    # Find directory where newest saved model is located
+    bucket = self._params['jobDir']
+    stats = gcs.listbucket(bucket[4:])
+    newest_file = None
+
+    for stat in stats:
+      if (stat.filename.find('saved_model.pb')) != -1:
```
[the remark this answers is not in the document]
Comment (LimNora) [reply backends/core/workers.py]: Thanks! I implemented this.

**Comment by LimNora** [comment]
Thanks a lot for the comments. I implemented the corrections.

**Pull request merged**